src/module.c

```diff
@@ -11244,6 +11442,30 @@ dictType sdsKeyValueHashDictType = {
     NULL                        /* allow to expand */
 };
 
+uint64_t dictModuleObjHash(const void *key) {
+    const RedisModule *o = key;
+    return dictGenHashFunction(o->name, sdslen((sds)o->name));
+}
+
+int dictModuleObjectCompare(dict *d, const void *key1, const void *key2) {
+    UNUSED(d);
+    const RedisModule *o1 = (RedisModule*) key1, *o2 = (RedisModule*) key2;
+    return sdscmp((sds) o1->name, (sds) o2->name);
+}
+
+/* Generic hash table type where keys are Redis Module Objects, Values
+ * are list of channel name (sds). */
+dictType moduleObjectKeyDictType = {
+        dictModuleObjHash,            /* hash function */
+        NULL,                      /* key dup */
+        NULL,                      /* val dup */
+        dictModuleObjectCompare,      /* key compare */
```

Comment:
Since `moduleChannelSubscribers` use the pointer of the module as key, so `dictModuleObjectKeyCompare` would be never entered.
the pointers of two modules are never equal, see this line: 
https://github.com/redis/redis/blob/46393f9819ae29ddb86cb9819d0b055a923037b4/src/dict.c#L678